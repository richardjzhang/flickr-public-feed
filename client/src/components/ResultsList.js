// @flow strict
/** @jsx jsx */
import { jsx } from '@emotion/core';
import styled from '@emotion/styled';
import * as React from 'react';
import Avatar from 'src/components/Avatar';
import {
  Card,
  CardHeader,
  CardThumbnail,
  CardContent,
} from 'src/components/Card';
import Checkbox from 'src/components/Checkbox';
import MasonryLayout from 'src/components/MasonryLayout';
import { HStack, HSpacer, VStack, VSpacer } from 'src/components/Stack';
import { BASE_UNIT, colors, fontSize, fontWeight } from 'src/styles';
import { getAuthor, getDate } from 'src/utils/flickr';
import { useWindowDimensions } from 'src/utils/hooks';

const Title = styled.div<{}>({
  fontSize: fontSize.large,
  fontWeight: fontWeight.semibold,
  color: colors.white,
});

const Description = styled.div<{}>({
  fontSize: fontSize.medium,
  color: colors.grey10,
});

function NoResults({ searchTerm }: {| searchTerm: string |}) {
  return (
    <VStack
      css={{
        padding: 8 * BASE_UNIT,
        textAlign: 'center',
      }}
    >
      <Title>
        Sorry, we couldn&#39;t find any matches for &#39;{searchTerm}&#39;
      </Title>
      <VSpacer height={4 * BASE_UNIT} />
      <Description>Please try searching with another term</Description>
    </VStack>
  );
}

// We will use the window resize as a measure for the number of columns
// rather than using a ref on MasonryLayout. This avoids all the nasty hacks
// we would need to perform to measure a ref's offset width before rendering
function getNumberOfColumns(windowWidth: number, windowColumnSize: number) {
  return Math.ceil(windowWidth / windowColumnSize);
}

const CARD_GAP = 5 * BASE_UNIT;
const CARD_WIDTH = 300;
const CardWrapper = styled.a<{}>({
  textDecoration: 'none',
  webkitAppearance: 'none',
  mozAppearance: 'none',
});

export default function ResultsList({
  searchTerm,
  results,
}: {|
  searchTerm: string,
  results: Array<any>,
|}) {
  const { width: windowWidth } = useWindowDimensions();
  const [numberOfCols, setNumberOfCols] = React.useState(
    getNumberOfColumns(windowWidth, 2 * CARD_WIDTH),
  );
  const [showTags, setShowTags] = React.useState(false);

  React.useEffect(() => {
    setNumberOfCols(
      Math.min(getNumberOfColumns(windowWidth, 2 * CARD_WIDTH), results.length),
    );
  }, [results, windowWidth]);

  if (results.length === 0) {
    return <NoResults searchTerm={searchTerm} />;
  }

  return (
    <VStack>
      <Checkbox
        value={showTags}
        onChange={() => setShowTags(!showTags)}
        text="Show tags"
      />
      <VSpacer height={4 * BASE_UNIT} />
      <MasonryLayout
        gap={CARD_GAP}
        numberOfCols={numberOfCols}
        colWidth={CARD_WIDTH}
      >
        {results.map((item) => {
          const author = getAuthor({ author: item.author });
          return (
            <CardWrapper key={item.link} href={item.link} target="_blank">
              <Card>
                <CardHeader>
                  <HStack>
                    <Avatar>{author[0].toUpperCase()}</Avatar>
                    <HSpacer width={16} />
                    <VStack>
                      <div css={{ color: colors.grey }}>{author}</div>
                      <VSpacer height={4} />
                      <div css={{ color: colors.grey90 }}>
                        {getDate({ published: item.published })}
                      </div>
                    </VStack>
                  </HStack>
                </CardHeader>
                <CardThumbnail
                  width={CARD_WIDTH}
                  src={item.media.m}
                  alt={item.title}
                />
                <CardContent>
                  <div>{item.title}</div>
                  {showTags && (
                    <React.Fragment>
                      <VSpacer height={3 * BASE_UNIT} />
                      <div
                        css={{
                          color: colors.grey90,
                          fontWeight: fontWeight.semibold,
                        }}
                      >
                        Tags:
                      </div>
                      <VSpacer height={BASE_UNIT} />
                      <div>{item.tags.split(' ').join(', ')}</div>
                    </React.Fragment>
                  )}
                </CardContent>
              </Card>
            </CardWrapper>
          );
        })}
      </MasonryLayout>
    </VStack>
  );
}
