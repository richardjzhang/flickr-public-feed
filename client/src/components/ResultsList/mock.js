// @flow strict

export const mockResults = [
  {
    author: 'nobody@flickr.com ("MartinZT")',
    author_id: '31011717@N03',
    date_taken: '2020-09-26T11:32:36-08:00',
    description:
      ' <p><a href="https://www.flickr.com/people/martinzt/">MartinZT</a> posted a photo:</p> <p><a href="https://www.flickr.com/photos/martinzt/50388183063/" title="2020_09_Salon_Prive_075"><img src="https://live.staticflickr.com/65535/50388183063_4eb10be552_m.jpg" width="240" height="160" alt="2020_09_Salon_Prive_075" /></a></p> ',
    link: 'https://www.flickr.com/photos/martinzt/50388183063/',
    media: {
      m: 'https://live.staticflickr.com/65535/50388183063_4eb10be552_m.jpg',
    },
    m: 'https://live.staticflickr.com/65535/50388183063_4eb10be552_m.jpg',
    published: '2020-09-27T09:23:01Z',
    tags: 'salonprivé salonprive astonmartin jaguar super car',
    title: '2020_09_Salon_Prive_075',
  },
  {
    author: 'nobody@flickr.com ("MartinZT")',
    author_id: '31011717@N03',
    date_taken: '2020-09-26T10:16:56-08:00',
    description:
      ' <p><a href="https://www.flickr.com/people/martinzt/">MartinZT</a> posted a photo:</p> <p><a href="https://www.flickr.com/photos/martinzt/50388197208/" title="2020_09_Salon_Prive_005"><img src="https://live.staticflickr.com/65535/50388197208_50c0bf8738_m.jpg" width="240" height="159" alt="2020_09_Salon_Prive_005" /></a></p> ',
    link: 'https://www.flickr.com/photos/martinzt/50388197208/',
    media: {
      m: 'https://live.staticflickr.com/65535/50388197208_50c0bf8738_m.jpg',
    },
    published: '2020-09-27T09:22:14Z',
    tags: 'salonprivé salonprive astonmartin jaguar super car',
    title: '2020_09_Salon_Prive_005',
  },
  {
    author: 'nobody@flickr.com ("marian.pile")',
    author_id: '183582427@N05',
    date_taken: '2020-09-26T15:37:50-08:00',
    description:
      ' <p><a href="https://www.flickr.com/people/183582427@N05/">marian.pile</a> posted a photo:</p> <p><a href="https://www.flickr.com/photos/183582427@N05/50388895711/" title="Swindon Supermarine v Tiverton Town"><img src="https://live.staticflickr.com/65535/50388895711_2703821a44_m.jpg" width="240" height="200" alt="Swindon Supermarine v Tiverton Town" /></a></p> <p></p>',
    link: 'https://www.flickr.com/photos/183582427@N05/50388895711/',
    media: {
      m: 'https://live.staticflickr.com/65535/50388895711_2703821a44_m.jpg',
    },
    published: '2020-09-27T09:27:31Z',
    tags:
      'away balls boots days footballers football goalkeeper goals grass keepers league matchday nonleague non players saturdays swindon super marine team tiverton town tivvy yellows z',
    title: 'Swindon Supermarine v Tiverton Town',
  },
  {
    author: 'nobody@flickr.com ("marian.pile")',
    author_id: '183582427@N05',
    date_taken: '2020-09-26T14:20:52-08:00',
    description:
      ' <p><a href="https://www.flickr.com/people/183582427@N05/">marian.pile</a> posted a photo:</p> <p><a href="https://www.flickr.com/photos/183582427@N05/50388896216/" title="Swindon Supermarine v Tiverton Town"><img src="https://live.staticflickr.com/65535/50388896216_d0509390fc_m.jpg" width="240" height="144" alt="Swindon Supermarine v Tiverton Town" /></a></p> <p></p>',
    link: 'https://www.flickr.com/photos/183582427@N05/50388896216/',
    media: {
      m: 'https://live.staticflickr.com/65535/50388896216_d0509390fc_m.jpg',
    },
    published: '2020-09-27T09:27:22Z',
    tags:
      'away balls boots days footballers football goalkeeper goals grass keepers league matchday nonleague non players saturdays swindon super marine team tiverton town tivvy yellows z',
    title: 'Swindon Supermarine v Tiverton Town',
  },
];
