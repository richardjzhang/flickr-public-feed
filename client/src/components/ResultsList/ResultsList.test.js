import { render } from '@testing-library/react';
import * as React from 'react';

import ResultsList from '.';
import { mockResults } from './mock';

describe('renders correctly', () => {
  test('renders no results when there is no search', () => {
    const tree = render(<ResultsList searchTerm="" results={[]} />);
    expect(tree).toMatchSnapshot();

    // There should be no columns present
    const firstColumn = tree.queryByTestId('MasonryLayout-Column-0');
    expect(firstColumn).not.toBeInTheDocument();

    // Empty list of results should return no results
    const noResults = tree.getByTestId('ResultsList-no-results');
    expect(noResults).toBeInTheDocument();
  });

  test('renders no results when search returns nothing', () => {
    const tree = render(<ResultsList searchTerm="weofiwe" results={[]} />);
    expect(tree).toMatchSnapshot();

    // There should be no columns present
    const firstColumn = tree.queryByTestId('MasonryLayout-Column-0');
    expect(firstColumn).not.toBeInTheDocument();

    // Empty list of results should return no results
    const noResults = tree.getByTestId('ResultsList-no-results');
    expect(noResults).toBeInTheDocument();
  });

  test('renders results when search returns something', () => {
    const tree = render(
      <ResultsList searchTerm="super" results={mockResults} />,
    );
    expect(tree).toMatchSnapshot();

    // There should be at least 1 column present
    const firstColumn = tree.getByTestId('MasonryLayout-Column-0');
    expect(firstColumn).toBeInTheDocument();

    // No results should not be in the document
    const noResults = tree.queryByTestId('ResultsList-no-results');
    expect(noResults).not.toBeInTheDocument();
  });
});
