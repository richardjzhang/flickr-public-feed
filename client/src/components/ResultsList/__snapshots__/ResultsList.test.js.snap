// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly renders no results when search returns nothing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1og332n-NoResults"
        data-testid="ResultsList-no-results"
      >
        <div
          class="css-pxqx0m"
        >
          Sorry, we couldn't find any matches for '
          weofiwe
          '
        </div>
        <div
          class="css-3scvxm-VSpacer"
        />
        <div
          class="css-epdson"
        >
          Please try searching with another term
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1og332n-NoResults"
      data-testid="ResultsList-no-results"
    >
      <div
        class="css-pxqx0m"
      >
        Sorry, we couldn't find any matches for '
        weofiwe
        '
      </div>
      <div
        class="css-3scvxm-VSpacer"
      />
      <div
        class="css-epdson"
      >
        Please try searching with another term
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders correctly renders no results when there is no search 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1og332n-NoResults"
        data-testid="ResultsList-no-results"
      >
        <div
          class="css-pxqx0m"
        >
          Sorry, we couldn't find any matches for '
          
          '
        </div>
        <div
          class="css-3scvxm-VSpacer"
        />
        <div
          class="css-epdson"
        >
          Please try searching with another term
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1og332n-NoResults"
      data-testid="ResultsList-no-results"
    >
      <div
        class="css-pxqx0m"
      >
        Sorry, we couldn't find any matches for '
        
        '
      </div>
      <div
        class="css-3scvxm-VSpacer"
      />
      <div
        class="css-epdson"
      >
        Please try searching with another term
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders correctly renders results when search returns something 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-ebhb4y-VStack"
      >
        <div
          class="css-1crab1l-Checkbox"
        >
          <input
            class="css-1xr1lj7"
            type="checkbox"
          />
          <div
            class="css-1rnx2nk-HSpacer"
          />
          <div
            class="css-u4h7m5"
          >
            Show tags
          </div>
        </div>
        <div
          class="css-3scvxm-VSpacer"
        />
        <div
          class="css-1a8sdri"
        >
          <div
            class="css-jnid37"
            data-testid="MasonryLayout-Column-0"
            width="300"
          >
            <a
              class="css-aicwwk"
              href="https://www.flickr.com/photos/martinzt/50388183063/"
              target="_blank"
            >
              <div
                class="css-1c535gx"
              >
                <div
                  class="css-9v8cyb"
                >
                  <div
                    class="css-1o4uhv2-HStack"
                  >
                    <div
                      class="css-1n49xm3"
                    >
                      M
                    </div>
                    <div
                      class="css-35lhvg-HSpacer"
                    />
                    <div
                      class="css-ebhb4y-VStack"
                    >
                      <div
                        class="css-14bxiw3"
                      >
                        MartinZT
                      </div>
                      <div
                        class="css-1uhajun-VSpacer"
                      />
                      <div
                        class="css-1ssvqp4"
                      >
                        27 Sep, 2020
                      </div>
                    </div>
                  </div>
                </div>
                <img
                  alt="2020_09_Salon_Prive_075"
                  class="css-1v5z18m"
                  src="https://live.staticflickr.com/65535/50388183063_4eb10be552_m.jpg"
                  width="300"
                />
                <div
                  class="css-1occhsy"
                >
                  <div>
                    2020_09_Salon_Prive_075
                  </div>
                </div>
              </div>
            </a>
            <div
              class="css-6gf715-VSpacer"
            />
            <a
              class="css-aicwwk"
              href="https://www.flickr.com/photos/183582427@N05/50388895711/"
              target="_blank"
            >
              <div
                class="css-1c535gx"
              >
                <div
                  class="css-9v8cyb"
                >
                  <div
                    class="css-1o4uhv2-HStack"
                  >
                    <div
                      class="css-1n49xm3"
                    >
                      M
                    </div>
                    <div
                      class="css-35lhvg-HSpacer"
                    />
                    <div
                      class="css-ebhb4y-VStack"
                    >
                      <div
                        class="css-14bxiw3"
                      >
                        marian.pile
                      </div>
                      <div
                        class="css-1uhajun-VSpacer"
                      />
                      <div
                        class="css-1ssvqp4"
                      >
                        27 Sep, 2020
                      </div>
                    </div>
                  </div>
                </div>
                <img
                  alt="Swindon Supermarine v Tiverton Town"
                  class="css-1v5z18m"
                  src="https://live.staticflickr.com/65535/50388895711_2703821a44_m.jpg"
                  width="300"
                />
                <div
                  class="css-1occhsy"
                >
                  <div>
                    Swindon Supermarine v Tiverton Town
                  </div>
                </div>
              </div>
            </a>
            <div
              class="css-6gf715-VSpacer"
            />
          </div>
          <div
            class="css-jnid37"
            data-testid="MasonryLayout-Column-1"
            width="300"
          >
            <a
              class="css-aicwwk"
              href="https://www.flickr.com/photos/martinzt/50388197208/"
              target="_blank"
            >
              <div
                class="css-1c535gx"
              >
                <div
                  class="css-9v8cyb"
                >
                  <div
                    class="css-1o4uhv2-HStack"
                  >
                    <div
                      class="css-1n49xm3"
                    >
                      M
                    </div>
                    <div
                      class="css-35lhvg-HSpacer"
                    />
                    <div
                      class="css-ebhb4y-VStack"
                    >
                      <div
                        class="css-14bxiw3"
                      >
                        MartinZT
                      </div>
                      <div
                        class="css-1uhajun-VSpacer"
                      />
                      <div
                        class="css-1ssvqp4"
                      >
                        27 Sep, 2020
                      </div>
                    </div>
                  </div>
                </div>
                <img
                  alt="2020_09_Salon_Prive_005"
                  class="css-1v5z18m"
                  src="https://live.staticflickr.com/65535/50388197208_50c0bf8738_m.jpg"
                  width="300"
                />
                <div
                  class="css-1occhsy"
                >
                  <div>
                    2020_09_Salon_Prive_005
                  </div>
                </div>
              </div>
            </a>
            <div
              class="css-6gf715-VSpacer"
            />
            <a
              class="css-aicwwk"
              href="https://www.flickr.com/photos/183582427@N05/50388896216/"
              target="_blank"
            >
              <div
                class="css-1c535gx"
              >
                <div
                  class="css-9v8cyb"
                >
                  <div
                    class="css-1o4uhv2-HStack"
                  >
                    <div
                      class="css-1n49xm3"
                    >
                      M
                    </div>
                    <div
                      class="css-35lhvg-HSpacer"
                    />
                    <div
                      class="css-ebhb4y-VStack"
                    >
                      <div
                        class="css-14bxiw3"
                      >
                        marian.pile
                      </div>
                      <div
                        class="css-1uhajun-VSpacer"
                      />
                      <div
                        class="css-1ssvqp4"
                      >
                        27 Sep, 2020
                      </div>
                    </div>
                  </div>
                </div>
                <img
                  alt="Swindon Supermarine v Tiverton Town"
                  class="css-1v5z18m"
                  src="https://live.staticflickr.com/65535/50388896216_d0509390fc_m.jpg"
                  width="300"
                />
                <div
                  class="css-1occhsy"
                >
                  <div>
                    Swindon Supermarine v Tiverton Town
                  </div>
                </div>
              </div>
            </a>
            <div
              class="css-6gf715-VSpacer"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-ebhb4y-VStack"
    >
      <div
        class="css-1crab1l-Checkbox"
      >
        <input
          class="css-1xr1lj7"
          type="checkbox"
        />
        <div
          class="css-1rnx2nk-HSpacer"
        />
        <div
          class="css-u4h7m5"
        >
          Show tags
        </div>
      </div>
      <div
        class="css-3scvxm-VSpacer"
      />
      <div
        class="css-1a8sdri"
      >
        <div
          class="css-jnid37"
          data-testid="MasonryLayout-Column-0"
          width="300"
        >
          <a
            class="css-aicwwk"
            href="https://www.flickr.com/photos/martinzt/50388183063/"
            target="_blank"
          >
            <div
              class="css-1c535gx"
            >
              <div
                class="css-9v8cyb"
              >
                <div
                  class="css-1o4uhv2-HStack"
                >
                  <div
                    class="css-1n49xm3"
                  >
                    M
                  </div>
                  <div
                    class="css-35lhvg-HSpacer"
                  />
                  <div
                    class="css-ebhb4y-VStack"
                  >
                    <div
                      class="css-14bxiw3"
                    >
                      MartinZT
                    </div>
                    <div
                      class="css-1uhajun-VSpacer"
                    />
                    <div
                      class="css-1ssvqp4"
                    >
                      27 Sep, 2020
                    </div>
                  </div>
                </div>
              </div>
              <img
                alt="2020_09_Salon_Prive_075"
                class="css-1v5z18m"
                src="https://live.staticflickr.com/65535/50388183063_4eb10be552_m.jpg"
                width="300"
              />
              <div
                class="css-1occhsy"
              >
                <div>
                  2020_09_Salon_Prive_075
                </div>
              </div>
            </div>
          </a>
          <div
            class="css-6gf715-VSpacer"
          />
          <a
            class="css-aicwwk"
            href="https://www.flickr.com/photos/183582427@N05/50388895711/"
            target="_blank"
          >
            <div
              class="css-1c535gx"
            >
              <div
                class="css-9v8cyb"
              >
                <div
                  class="css-1o4uhv2-HStack"
                >
                  <div
                    class="css-1n49xm3"
                  >
                    M
                  </div>
                  <div
                    class="css-35lhvg-HSpacer"
                  />
                  <div
                    class="css-ebhb4y-VStack"
                  >
                    <div
                      class="css-14bxiw3"
                    >
                      marian.pile
                    </div>
                    <div
                      class="css-1uhajun-VSpacer"
                    />
                    <div
                      class="css-1ssvqp4"
                    >
                      27 Sep, 2020
                    </div>
                  </div>
                </div>
              </div>
              <img
                alt="Swindon Supermarine v Tiverton Town"
                class="css-1v5z18m"
                src="https://live.staticflickr.com/65535/50388895711_2703821a44_m.jpg"
                width="300"
              />
              <div
                class="css-1occhsy"
              >
                <div>
                  Swindon Supermarine v Tiverton Town
                </div>
              </div>
            </div>
          </a>
          <div
            class="css-6gf715-VSpacer"
          />
        </div>
        <div
          class="css-jnid37"
          data-testid="MasonryLayout-Column-1"
          width="300"
        >
          <a
            class="css-aicwwk"
            href="https://www.flickr.com/photos/martinzt/50388197208/"
            target="_blank"
          >
            <div
              class="css-1c535gx"
            >
              <div
                class="css-9v8cyb"
              >
                <div
                  class="css-1o4uhv2-HStack"
                >
                  <div
                    class="css-1n49xm3"
                  >
                    M
                  </div>
                  <div
                    class="css-35lhvg-HSpacer"
                  />
                  <div
                    class="css-ebhb4y-VStack"
                  >
                    <div
                      class="css-14bxiw3"
                    >
                      MartinZT
                    </div>
                    <div
                      class="css-1uhajun-VSpacer"
                    />
                    <div
                      class="css-1ssvqp4"
                    >
                      27 Sep, 2020
                    </div>
                  </div>
                </div>
              </div>
              <img
                alt="2020_09_Salon_Prive_005"
                class="css-1v5z18m"
                src="https://live.staticflickr.com/65535/50388197208_50c0bf8738_m.jpg"
                width="300"
              />
              <div
                class="css-1occhsy"
              >
                <div>
                  2020_09_Salon_Prive_005
                </div>
              </div>
            </div>
          </a>
          <div
            class="css-6gf715-VSpacer"
          />
          <a
            class="css-aicwwk"
            href="https://www.flickr.com/photos/183582427@N05/50388896216/"
            target="_blank"
          >
            <div
              class="css-1c535gx"
            >
              <div
                class="css-9v8cyb"
              >
                <div
                  class="css-1o4uhv2-HStack"
                >
                  <div
                    class="css-1n49xm3"
                  >
                    M
                  </div>
                  <div
                    class="css-35lhvg-HSpacer"
                  />
                  <div
                    class="css-ebhb4y-VStack"
                  >
                    <div
                      class="css-14bxiw3"
                    >
                      marian.pile
                    </div>
                    <div
                      class="css-1uhajun-VSpacer"
                    />
                    <div
                      class="css-1ssvqp4"
                    >
                      27 Sep, 2020
                    </div>
                  </div>
                </div>
              </div>
              <img
                alt="Swindon Supermarine v Tiverton Town"
                class="css-1v5z18m"
                src="https://live.staticflickr.com/65535/50388896216_d0509390fc_m.jpg"
                width="300"
              />
              <div
                class="css-1occhsy"
              >
                <div>
                  Swindon Supermarine v Tiverton Town
                </div>
              </div>
            </div>
          </a>
          <div
            class="css-6gf715-VSpacer"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
